From 23c070ce882cafb5cf9a6298e26fdd386eab218d Mon Sep 17 00:00:00 2001
From: Mark Hibberd <mark@hibberd.id.au>
Date: Tue, 5 Jun 2018 20:47:38 +1000
Subject: [PATCH] Fix OS linking issues.

---
 main/mafia.hs | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/main/mafia.hs b/main/mafia.hs
index 73ace38..50e2bf8 100644
--- a/main/mafia.hs
+++ b/main/mafia.hs
@@ -43,6 +43,7 @@ import           Mafia.P
 import           System.Environment (getArgs)
 import           System.IO (BufferMode(..), hSetBuffering)
 import           System.IO (IO, FilePath, stdout, stderr)
+import qualified System.Info as Info
 
 import           Control.Monad.Trans.Either (EitherT, hoistEither, left)
 import           Control.Monad.Trans.Bifunctor (firstT, bimapT)
@@ -513,7 +514,18 @@ mafiaBuild p w dumpc dumpa flags args = do
           ,"-ddump-to-file"
           ]
 
-  liftCabal . cabal_ "build" $ ["-j"] <> wargs <> dumpargs <> args
+    platformargs =
+      case Info.os of
+        "darwin" -> [
+            "--ghc-options=-optl-Wl,-dead_strip_dylibs"
+          , "--ghc-options=-optc-Wno-unused-command-line-argument"
+          , "--ghc-options=-optl-Wno-unused-command-line-argument"
+          ]
+        _ -> [
+          ]
+
+
+  liftCabal . cabal_ "build" $ ["-j"] <> wargs <> dumpargs <> platformargs <> args
 
 mafiaTest :: [Flag] -> [Argument] -> EitherT MafiaError IO ()
 mafiaTest flags args = do

